planetaryimage/image.py

```diff
@@ -26,10 +29,25 @@ def open(cls, filename):
         filename : string
             name of file to read as an image file
         """
-        with open(filename, 'rb') as fp:
-            return cls(fp, filename)
+        if filename.endswith('.gz'):
+            try:
+                with gzip.open(filename, 'rb') as fp:
+                    return cls(fp, filename, compression='gz')
+            except AttributeError:
+                raise IOError('GZIP compression not supported until Python 2.6.')
+                return
+        elif filename.endswith('.bz2'):
+            try:
+                with bz2.open(filename, 'rb') as fp:
```

Comment:
Oh, I never thought of using `finally`.  That seems like a good approach.  I thought we would have to track down all of the right places to close the stream and that seemed like a bad idea.